src/monitors/R3Monitor.v3

```diff
@@ -0,0 +1,572 @@
+// Copyright 2024 Wizard Authors. All rights reserved.
+// See LICENSE for details of Apache 2.0 license.
+
+def E_ = MonitorRegistry.add(
+	"r3", "Generates a reduced set of event traces for wasm r3 replay module creation.",
+	R3Monitor.new());
+
+var ticks: void -> u64;
+var filter: DeclFilter;
+class R3Monitor extends Monitor {
+	def var handler: EventHandler;
+
+	def onParse(module: Module) {
+		handler = EventHandler.new(module);
+		Instrumentation.insertGlobalProbe(ReturnProbe.new(handler));
+		var mm = ModuleInstrumenter.new(module);
+		var visitor = R3MonitorBytecodeVisitor.new(handler, module, BytecodeIterator.new());
+		mm.forEachFunc(instrumentFunc(visitor, _, _));
+		mm.beforeMemRead(handler.memoryRead);
+		mm.beforeMemWrite(handler.memoryWrite);
+		mm.beforeMemGrow(handler.memoryGrow);
```

Comment:
We'll also need to instrument `memory.init`, `memory.copy`, and `memory.fill`.

Reply:
done